### ISSUE TITLE
iOS and Android: Consistent messaging in Publicize to Facebook

### ISSUE THREAD

**Issue by shiki** [wordpress-mobile/WordPress-Android#10278]
This covers both iOS and Android but we're mainly tracking it here in this Android issue.

## Problem

In p7cLQ7-DY-p2, we found that there are inconsistencies in how the mobile apps present information regarding the requirements for successfully connecting a Facebook account.

Connecting a Facebook account can be achieved by navigating to a site and tapping on Sharing → Facebook.

Here is a connection flow diagram that shows the inconsistencies:

![Facebook@2x(1)](https://user-images.githubusercontent.com/198826/61804227-c9265680-adf0-11e9-8891-f2acaa8afddd.png)

- iOS does not show any requirements information on the first page but shows it after the user fails to connect.
- Android shows some requirements information on the first page but the link is pointing to an old blog post, not the support page.

### Connection Requirements

The known requirements to successfully connect a Facebook account are:

1. The user should have a Facebook Page.
2. The user should select a Facebook Page when going through the connection flow.
3. The Facebook Page should be set to [**Published**](https://www.facebook.com/help/184605634921611).

This information is displayed on our support page. However, there is no mention of it in the apps.

## Solution

@thehenrybyrd, @rfaile313, and I propose to make these changes.

### Copy changes

On the first page, the description can be modified slightly to describe the requirement for Pages to be published.

> Connect to automatically share your blog posts to a Published Facebook Page

The error message shown after failing to connect can be made more concise and mention the requirements.

> We couldn’t connect this Facebook account because it doesn’t have access to any Published Facebook Pages. Facebook allows automatic sharing only to Pages which are Published.

### Android Flow

The first page on Android shows outdated information. We propose to change the flow on Android to match iOS instead. This wireframe shows the desired flow as well as the proposed copy changes.

![Android Flow@2x(1)](https://user-images.githubusercontent.com/198826/61806301-b281fe80-adf4-11e9-9b1e-a9fac3b70d2b.png)

## Editorial Team

We'd love to have your input on the copy changes we're proposing.

## Design Team

We'd love to hear your input on this as well. Does the existing iOS design still match our current standards? Or is there something that should be changed?

## Tasks

- [x] Wait for input from Editorial and Design 
- [x] Create an issue for iOS https://github.com/wordpress-mobile/WordPress-iOS/issues/12228
- [x] Implement on iOS. Done in https://github.com/wordpress-mobile/WordPress-iOS/pull/12337
- [ ] Implement on Android

**Comment by mattmiklic** [comment]
I think we could actually simplify this a bit to standardize on what Android is doing rather than iOS. For a couple of reasons:

- a dialog (Android) or alert (iOS) is the standard way to handle an error message that should be acknowledged by the user before continuing.
- Using a full-screen modal for the error state feels like it's over-emphasized, and the illustration is superfluous.
- We can align the copy with what's now being used on WordPress.com.
- I've used "No Pages Found" as a title for the dialog/alert so it's consistent between platforms (since alerts should have a title on iOS), but maybe Editorial could weigh in with a better title.

Android:
<img width="624" alt="Screen Shot 2019-07-24 at 1 45 13 PM" src="https://user-images.githubusercontent.com/1396552/61816116-afe5d000-ae19-11e9-85dd-626817a378a0.png">

iOS:
<img width="645" alt="Screen Shot 2019-07-24 at 1 45 08 PM" src="https://user-images.githubusercontent.com/1396552/61816121-b2e0c080-ae19-11e9-833a-1ea54ca0de6e.png">

**Comment by mattmiklic** [comment]
I agree that the the current "The facebook connection could not be made because..." message is too wordy, though. Ideally it would just be a sentence or two. Saying "we" couldn't connect feels a little strange when it's an app saying it, but I defer to Editorial's guidance on how we could make that message more concise.

**Comment by carolynannewells** [comment]
Hello everyone! Chipping in from Editorial - thanks for pinging us! 

I would perhaps take out the capitalization of 'Published': 

Connect to automatically share your blog posts to a published Facebook Page.

Error message: 

The Facebook connection cannot find any Pages. Publicize cannot connect to Facebook Profiles, only Pages. 

Instead of 'No Pages Found' we could perhaps use something a little friendlier such as: 

Something went wrong!

Oh no, missing Pages! 

Pages won’t reply!

**Comment by shiki** [comment]
@carolynannewells Thank you for responding!

Regarding the error message you suggested:

> The Facebook connection cannot find any Pages. Publicize cannot connect to Facebook Profiles, only Pages.

In the description above, https://github.com/wordpress-mobile/WordPress-Android/issues/10278#issue-472347068, the proposed error message mentioned the "published" requirement:

> We couldn’t connect this Facebook account because it doesn’t have access to any Published Facebook Pages. Facebook allows automatic sharing only to Pages which are Published.

Do you think that's no longer necessary in this context?

**Comment by carolynannewells** [comment]
We can add the word published back in to make that clearer:

The Facebook connection cannot find any Pages. Publicize cannot connect to Facebook Profiles, only published Pages.

**Comment by shiki** [comment]
Thank you for the quick feedback, @mattmiklic and @carolynannewells!

**Comment by cerezo074** [comment]
Hi there, I am working actually this enhancement for ios project and I have a question. Where are located the copies for the other languages? I've sought here [https://translate.wordpress.org/projects/apps/ios/dev/](Translations) but I've couldn't find any resource for the alert title(**No Pages Found**). This is my first contribution, my apologies if I've missed something.

**Comment by mattmiklic** [comment]
> The positions for them depends on the selected style as you see, i've applied the same style as you wish but the "OK" shows first(this is done by the OS) because the styles has priority.

@cerezo074 That's just fine. I didn't realize the button ordering was set by the system, that's good to know.

For the message text, let's use what @carolynannewells suggested above. For the title I believe the clearest possible message would be "Not connected" -- we want to let the user know as clearly as possible that the Facebook connection they just attempted couldn't be made. That title would do it.

So we should have something that looks like:

<img width="630" alt="Screen Shot 2019-08-19 at 10 56 07 AM" src="https://user-images.githubusercontent.com/1396552/63275712-035c0a00-c270-11e9-9156-cd03c616527b.png">

**Comment by shiki** [comment]
@cerezo074 

> 1. Where are located the copies for the other languages?

You shouldn't need to worry about other languages. As long as you use `NSLocalizedString()` [like this](https://github.com/wordpress-mobile/WordPress-iOS/blob/d8f58a115b0cf83a93f21966c91010323c44a874/WordPress/Classes/ViewRelated/Domains/Register/RegisterDomainDetails/ViewController/RegisterDomainDetailsViewController%2BLocalizedStrings.swift#L5-L8), the translators will take care of the rest.

**Comment by jd-alexander** [comment]
@shiki I am looking into this issue and I just wanted to know if the UI of Android should be changed to match iOS in design style or the current design can remain and the flow is just made identical? 

The only new design would be error page right?

**Comment by mattmiklic** [comment]
@jd-alexander my recommendation [above](https://github.com/wordpress-mobile/WordPress-Android/issues/10278?email_token=AAKU6SHYGEG6ZMNUHPCTAHLQBDECTA5CNFSM4IGRHLZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2XWHZY#issuecomment-514732564) was to keep Android the same, and switch to using a modal dialog to display the error on iOS.

**Comment by jd-alexander** [comment]
Thanks @mattmiklic I was super focused on the original issue description when I asked that. Thanks for pointing this out. I appreciate it 🙏
